api/tests/routes/pro/signin_test.py

```diff
@@ -133,7 +135,8 @@ def test_whith_user_offerer(self, client):
         # 2. stamp session
         # 3. fetch user for serialization
         # 4. fetch user offerer
-        with assert_num_queries(4):
+        # 4. fetch user pro nav state
```

Comment:
On pourrait `joinedload` l'activation de la nouvelle navigation, vu que c'est une relation one to one c'est gratuit et ça fait une query de moins

C'est dans `src/pcapi/core/user/repository.py` à la ligne 64

Reply:
@damien-ramelet : la fonction que tu mentionnes (`_find_user_by_email_query()` est certainement utilisée à d'autres endroits que pour les routes qui nous intéressent ici. Y ajouter un `joinedload` a donc probablement un coût (sur tous les autres usages) qui dépasse le gain qu'on aurait à ne pas faire de requête SQL supplémentaire lors du login d'un acteur culturel. Il faudrait soit ajouter le `joinedload` sur la (ou les) fonction(s) spécifiquement utilisée(s) par les routes qui nous intéressent ici... ou, probablement, vivre avec cette requête SQL supplémentaire si ça n'est pas possible.

À vue de nez, les routes qui nous intéressent ici sont : 
- `/signin`, qui utilise `get_user_with_credentials`, également utilisé pour l'authentification sur l'app bénéficiaires
- `/users/current` qui utilise `pcapi.utils.login_manager.get_user_with_id`, qui est également utilisée par toutes les routes du portail pro (sauf celle d'authentification).

Il me semble donc embêtant d'ajouter `joinedload` dans ces fonctions, car ça ajouterait le coût de la jointure (faible, certes) sur toutes les routes. Je pense qu'on peut vivre avec la requête SQL supplémentaire. :)

Reply:
J'ai pas regardé au delà de l'usage de `_find_user_by_email_query`, mais effectivement les fonctions qui l'appelle sont utilisées assez massivement dans le repo. Je resolve !